distribution/lib/Standard/Visualization/0.0.0-dev/src/Table/Visualization.enso

```diff
@@ -191,28 +191,122 @@ make_json_for_xml_element xml_element max_items type:Text="XML_Element" =
      to display.
    - all_rows_count: the number of all rows in the underlying data, useful if
      only a fragment is displayed.
-make_json_for_table : Table -> Integer -> Integer -> Boolean -> Boolean -> JS_Object
-make_json_for_table dataframe max_rows all_rows_count include_index_col is_db_table =
+make_json_for_table : Table -> Integer -> Integer -> Boolean -> JS_Object
+make_json_for_table dataframe max_rows all_rows_count is_db_table =
     act_max = if max_rows < all_rows_count then max_rows else all_rows_count
     get_vector c         = Warning.set (Vector.new act_max i-> make_json_for_value (c.get i)) []
     columns              = dataframe.columns
     header               = ["header", columns.map .name]
     value_type           = ["value_type", columns.map .value_type]
-    data                 = ["data",   columns.map get_vector]
     all_rows             = ["all_rows_count", all_rows_count]
-    has_index_col        = ["has_index_col", include_index_col]
+    has_index_col        = ["has_index_col", True]
+    is_ssrm              = if is_db_table then False else all_rows_count > max_rows
     links                = ["get_child_node_action", "get_row"]
     child_label          = ["child_label", "row"]
+    data                 = if is_ssrm then Nothing else columns.map get_vector
     data_quality_metrics = if is_db_table then [] else
         number_nothing = JS_Object.from_pairs [["name", "Count nothings"], ["percentage_value", columns.map .count_nothing]]
         sampled_label = if all_rows_count > Column.default_sample_size then " (sampled)" else ""
         number_untrimmed = JS_Object.from_pairs [["name", "Count untrimmed whitespace" + sampled_label], ["percentage_value", columns.map .count_untrimmed]]
         number_non_triv = JS_Object.from_pairs [["name", "Count non trivial whitespace" + sampled_label], ["percentage_value", columns.map .count_non_trivial_whitespace]]
         JS_Object.from_pairs
         [number_nothing, number_untrimmed, number_non_triv]
-    pairs                = [header, value_type, data, all_rows, has_index_col, links, ["data_quality_metrics", data_quality_metrics] ,["type", "Table"], child_label]
+    pairs                = [header, value_type, ["data", data], all_rows, has_index_col, links, ["data_quality_metrics", data_quality_metrics] ,["type", "Table"], child_label,  ["is_ssrm", is_ssrm]]
     JS_Object.from_pairs pairs
 
+## PRIVATE
+
+   applies the sort model to the table.
+
+   Arguments:
+   - start_number: the start index of the next 100 rows to get
+   - sort_col_index_list: indexes of columns with sorts applies
+   - sort_direction_list: list of sort directions 1 for ascending -1 for descending
+
+get_distinct_values_for_column : Table -> Integer -> JS_Object
+get_distinct_values_for_column x column_index =
```

Comment:
The doc comment is not aligned with the method that it is documenting, perhaps copied from `apply_sort_to_table`.